ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -0,0 +1,244 @@
+import io.confluent.ksql.util.SchemaUtil;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.LongSupplier;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Schema.Type;
+
+public class InsertValuesExecutor {
+
+  private static final long MAX_SEND_TIMEOUT_SECONDS = 5;
+
+  private final LongSupplier clock;
+
+  public InsertValuesExecutor() {
+    this(System::currentTimeMillis);
+  }
+
+  @VisibleForTesting
+  InsertValuesExecutor(final LongSupplier clock) {
+    this.clock = Objects.requireNonNull(clock, "clock");
+  }
+
+  public Optional<KsqlEntity> execute(
+      final ConfiguredStatement<InsertValues> statement,
+      final KsqlExecutionContext executionContext,
+      final ServiceContext serviceContext
+  ) {
+    final InsertValues insertValues = statement.getStatement();
+    final StructuredDataSource<?> dataSource = executionContext
+        .getMetaStore()
+        .getSource(insertValues.getTarget().getSuffix());
+
+    if (dataSource instanceof KsqlStream && ((KsqlStream<?>) dataSource).hasWindowedKey()) {
+      throw new KsqlException("INSERT VALUES does not support windowed data sources!");
+    }
+
+    final KsqlConfig config = statement.getConfig()
+        .cloneWithPropertyOverwrite(statement.getOverrides());
+
+    final GenericRow row = extractRow(insertValues, dataSource);
+    final byte[] key = serializeKey(row.getColumnValue(SchemaUtil.ROWKEY_NAME_INDEX), dataSource);
+    final byte[] value = serializeRow(row, dataSource, config, serviceContext);
+
+    final String topicName = dataSource.getKafkaTopicName();
+
+    // for now, just create a new producer each time
+    final Producer<byte[], byte[]> producer = serviceContext
+        .getKafkaClientSupplier()
+        .getProducer(config.getProducerClientConfigProps());
+
+    producer.send(
+        new ProducerRecord<>(
+            topicName,
+            null,
+            row.getColumnValue(SchemaUtil.ROWTIME_INDEX),
+            key,
+            value
+        )
+    );
+
+    producer.close(Duration.ofSeconds(MAX_SEND_TIMEOUT_SECONDS));
+
+    return Optional.empty();
+  }
+
+  private GenericRow extractRow(
+      final InsertValues insertValues,
+      final StructuredDataSource<?> dataSource
+  ) {
+    final Optional<String> keyField = dataSource.getKeyField().name();
+
+    final List<String> columns = insertValues.getColumns().isEmpty()
+        ? dataSource.getSchema()
+          .fields()
+          .stream()
+          .map(Field::name)
+          .filter(name -> !SchemaUtil.ROWTIME_NAME.equals(name))
```

Comment:
I thought we allow setting ROWTIME!

Reply:
We do! We just require you to specify it in the schema (we decided that the default schema does not include ROWTIME):
```
ksql> INSERT INTO PAGEVIEWS (ROWTIME, ROWKEY, USERID) VALUES (123, '2', 'Almog');
ksql> SELECT * FROM PAGEVIEWS;
...
123 | 2 | null | null | Almog
```